### PR TITLE
KAFKA-17491: Move BrokerServerMetrics to server module

### PR DESCRIPTION
**Pull request by wernerdv**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by wernerdv** [comment]
@chia7712 @mimaison Hello, please take a look at the PR.

File: server/src/main/java/org/apache/kafka/server/metrics/BrokerServerMetrics.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                "metadata-apply-error-count",
+                METRIC_GROUP_NAME,
+                "The number of errors encountered by the BrokerMetadataPublisher while applying a new MetadataImage based on the latest MetadataDelta."
+        );
+
+        addMetric(metrics, lastAppliedRecordOffsetName, ignored -> lastAppliedImageProvenance.get().lastContainedOffset());
+        addMetric(metrics, lastAppliedRecordTimestampName, ignored -> lastAppliedImageProvenance.get().lastContainedLogTimeMs());
+        addMetric(metrics, lastAppliedRecordLagMsName, now -> now - lastAppliedImageProvenance.get().lastContainedLogTimeMs());
+        addMetric(metrics, metadataLoadErrorCountName, ignored -> metadataLoadErrorCount.get());
+        addMetric(metrics, metadataApplyErrorCountName, ignored -> metadataApplyErrorCount.get());
```
Comment (mimaison) [inline server/src/main/java/org/apache/kafka/server/metrics/BrokerServerMetrics.java]: For these 5 lines, could we do:
```
metrics.addMetric(lastAppliedRecordOffsetName, (config, now) -> lastAppliedImageProvenance.get().lastContainedOffset());
metrics.addMetric(lastAppliedRecordTimestampName, (config, now) -> lastAppliedImageProvenance.get().lastContainedLogTimeMs());
metrics.addMetric(lastAppliedRecordLagMsName, (config, now) -> now - lastAppliedImageProvenance.get().lastContainedLogTimeMs());
metrics.addMetric(metadataLoadErrorCountName, (config, now) -> metadataLoadErrorCount.get());
metrics.addMetric(metadataApplyErrorCountName, (config, now) -> metadataApplyErrorCount.get());
```
and not need the `FuncGauge` class and the `addMetrics()` method?

**Review by mimaison — COMMENTED** [review]
Thanks for the PR! It looks good overall, I just left a question.

File: server/src/main/java/org/apache/kafka/server/metrics/BrokerServerMetrics.java (hunk 1 above)
Comment (wernerdv) [reply server/src/main/java/org/apache/kafka/server/metrics/BrokerServerMetrics.java]: that's a good point, done.

**Review by mimaison — APPROVED** [review]
LGTM, thanks for the PR

**Pull request merged**